### PR TITLE
cleanup: switching to stubserver in tests instead of testservice implementation

### PR DESCRIPTION
**Pull request by janardhanvissa**
Partially Addresses: #7291 

RELEASE NOTES: None

### CODE REVIEW

**Comment by purnesh42H** [comment]
Can you mention the tests in description you are modifying?

File: orca/service_test.go
Comment (purnesh42H) [inline orca/service_test.go:115]: nit: stubServer

File: orca/service_test.go
Comment (purnesh42H) [inline orca/service_test.go:117]: why change the timeout?

Comment (purnesh42H) [reply orca/service_test.go:117]: Also, defaultTestTimeout is usually 10 secs

File: orca/service_test.go
Comment (purnesh42H) [inline orca/service_test.go:140]: why is this moved out of for loop?

File: orca/service_test.go
Comment (purnesh42H) [inline orca/service_test.go:157]: is this change related? Let's avoid unrelated changes

File: orca/service_test.go
Comment (janardhanvissa) [reply orca/service_test.go:117]: The default is 5 secs, getting context deadline issues. So, updated to 10 secs.

File: test/xds/xds_server_certificate_providers_test.go
Comment (purnesh42H) [inline test/xds/xds_server_certificate_providers_test.go:272]: instead of deleting this, you can simply edit to "and it set it on a stub server"

Comment (purnesh42H) [reply test/xds/xds_server_certificate_providers_test.go:272]: can remove the below line of setting the server and starting service

File: test/xds/xds_server_integration_test.go
Comment (purnesh42H) [inline test/xds/xds_server_integration_test.go:110]: same comment here. Just modify the existing statement instead of writing a new line

File: test/xds/xds_server_serving_mode_test.go
Comment (purnesh42H) [inline test/xds/xds_server_serving_mode_test.go:209]: same here

File: test/xds/xds_server_serving_mode_test.go
Comment (purnesh42H) [inline test/xds/xds_server_serving_mode_test.go:65]: same here

File: test/xds/xds_server_serving_mode_test.go
Comment (purnesh42H) [inline test/xds/xds_server_serving_mode_test.go:217]: s/Initialize an xDS-enabled gRPC server and use the helper to start the test service/Initialize an xDS-enabled gRPC server and sets it on stub server.

Comment (janardhanvissa) [reply test/xds/xds_server_serving_mode_test.go]: Updated the comments and pushed the changes.

**Review by purnesh42H — APPROVED** [review]
lgtm

**Comment by purnesh42H** [comment]
You need to rebase to latest main branch to unblock from vet check failure

**Comment by purnesh42H** [comment]
adding @easwars for second review

File: test/xds/xds_server_certificate_providers_test.go
Comment (purnesh42H) [inline test/xds/xds_server_certificate_providers_test.go:276]: s/sets it on a stub server/assigned to a stub server

Comment (janardhanvissa) [reply test/xds/xds_server_certificate_providers_test.go]: Addressed the comments and pushed the changes. Please re-review once.

**Review by easwars — APPROVED** [review]

**Pull request merged**
